src/NuGetPackageVerifier/PackageIssueFactory.cs

```diff
@@ -38,6 +39,15 @@ public static PackageVerifierIssue AssemblyMissingServicingAttribute(string asse
                 PackageIssueLevel.Error);
         }
 
+        public static PackageVerifierIssue AssemblyVersionDoesNotMatchPackageVersion(string currentFilePath, Version assemblyVersion, Version packageVersion, string packageId)
+        {
+            return new PackageVerifierIssue(
+                "ASSEMBLY_VERSION_MISMATCH",
+                currentFilePath,
```

Comment:
Could just name this `assemblyPath` to maintain consistency with other methods in this class.